sqlmesh/core/engine_adapter/risingwave.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+import logging
+import typing as t
+
+
+from sqlglot import exp, Dialect
+
+from sqlmesh.core.engine_adapter.base_postgres import BasePostgresEngineAdapter
+from sqlmesh.core.engine_adapter.mixins import (
+    GetCurrentCatalogFromFunctionMixin,
+    PandasNativeFetchDFSupportMixin,
+)
+from sqlmesh.core.engine_adapter.shared import (
+    set_catalog,
+    CatalogSupport,
+    CommentCreationView,
+    CommentCreationTable,
+)
+from sqlmesh.core.schema_diff import SchemaDiffer
+
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core._typing import SessionProperties
+    from sqlmesh.core.engine_adapter._typing import DF
+
+logger = logging.getLogger(__name__)
+
+
+@set_catalog()
+class RisingwaveEngineAdapter(
+    BasePostgresEngineAdapter,
+    PandasNativeFetchDFSupportMixin,
+    GetCurrentCatalogFromFunctionMixin,
+):
+    DIALECT = "risingwave"
+    SUPPORTS_INDEXES = True
+    HAS_VIEW_BINDING = True
+    CURRENT_CATALOG_EXPRESSION = exp.column("current_catalog")
+    SUPPORTS_REPLACE_TABLE = False
+    DEFAULT_BATCH_SIZE = 400
+    CATALOG_SUPPORT = CatalogSupport.SINGLE_CATALOG_ONLY
+    COMMENT_CREATION_TABLE = CommentCreationTable.COMMENT_COMMAND_ONLY
+    COMMENT_CREATION_VIEW = CommentCreationView.COMMENT_COMMAND_ONLY
+    SUPPORTS_MATERIALIZED_VIEWS = True
+
+    SCHEMA_DIFFER = SchemaDiffer(
```

Comment:
This appears to be copy+pasted directly from the Postgres adapter. Rather than duplicating it, can you extend it instead?